### ISSUE TITLE
[copy] Shorthand relative path causes go to panic

### ISSUE THREAD

**Issue by williexu** [Azure/azure-storage-azcopy#222]
### Which version of the AzCopy was used? 
10.0.7-Preview

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows

### What command did you run? 
`copy https://wilxstorage.blob.core.windows.net/container1/blob1 dkfjskj `
### What problem was encountered?
```
Scanning...
AZCOPY_OAUTH_TOKEN_INFO is set.
Using OAuth token for authentication.

Job 854c2de1-204c-2c43-4f8e-3841367ebe06 has started

Log file is located at: C:\Users\wilx/.azcopy/854c2de1-204c-2c43-4f8e-3841367ebe06.log
0 Done, 0 Failed, 1 Pending, 0 Skipped, 1 Totalpanic: runtime error: slice bounds out of range

goroutine 338 [running]:
github.com/Azure/azure-storage-azcopy/common.CreateParentDirectoryIfNotExist(0xc042b6a168, 0x7, 0xc04208bd10, 0xc042a8fbd0)
        /go/src/github.com/Azure/azure-storage-azcopy/common/writeThoughFile.go:30 +0x128
github.com/Azure/azure-storage-azcopy/common.CreateFileOfSizeWithWriteThroughOption(0xc042b6a168, 0x7, 0xd2e, 0x0, 0xd2e, 0xc042b6a168, 0x7)
        /go/src/github.com/Azure/azure-storage-azcopy/common/writeThoughFile_windows.go:34 +0x40
github.com/Azure/azure-storage-azcopy/ste.remoteToLocal(0xbd8c60, 0xc042ac80a0, 0xbc5ce0, 0xc04208bcc0, 0xc04201a2c0, 0xb71968)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/xfer-remoteToLocal.go:80 +0x298
github.com/Azure/azure-storage-azcopy/ste.parameterizeDownload.func1(0xbd8c60, 0xc042ac80a0, 0xbc5ce0, 0xc04208bcc0, 0xc04201a2c0)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/xfer.go:63 +0x6a
github.com/Azure/azure-storage-azcopy/ste.(*jobPartMgr).StartJobXfer(0xc046afc000, 0xbd8c60, 0xc042ac80a0)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/mgr-JobPartMgr.go:438 +0x6c
github.com/Azure/azure-storage-azcopy/ste.(*jobPartTransferMgr).StartJobXfer(0xc042ac80a0)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/mgr-JobPartTransferMgr.go:113 +0x4e
github.com/Azure/azure-storage-azcopy/ste.(*jobsAdmin).transferProcessor.func1(0xbd8c60, 0xc042ac80a0)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/JobsAdmin.go:251 +0x1ae
github.com/Azure/azure-storage-azcopy/ste.(*jobsAdmin).transferProcessor(0xc0421080e0, 0x1f)
        /go/src/github.com/Azure/azure-storage-azcopy/ste/JobsAdmin.go:259 +0x8a
created by github.com/Azure/azure-storage-azcopy/ste.initJobsAdmin
        /go/src/github.com/Azure/azure-storage-azcopy/ste/JobsAdmin.go:180 +0x66d
```

### How can we reproduce the problem in the simplest way?
Run the copy command and download a blob while indicating a file within the same directory without referencing the current directory using `./`

### Have you found a mitigation/solution?
Yes, use:  `copy https://wilxstorage.blob.core.windows.net/container1/blob1 ./dkfjskj `

**Comment by zezha-msft** [comment]
Hi @williexu, thanks for letting us know!

Currently, we do not support shorthand for relative paths. We could consider adding this support. And we should fix the error message.

**Comment by williexu** [comment]
@zezha-msft the command `copy Dockerfile https://wilxstorage.blob.core.windows.net/container1/blob2 ` works fine. 
It looks like shorthand relative paths work for the source, but not the destination.

This is pretty strange, as I'd expect consistency between the two.

**Comment by seguler** [comment]
>  It looks like shorthand relative paths work for the source, but not the destination.

Shouldn't this be a bug then ?

**Comment by zezha-msft** [comment]
@seguler we never supported shorthands. It's possible that the shell is replacing the shorthand with a path. 

In general though, isn't it better to be explicit by using a relative path?

**Comment by williexu** [comment]
I don't believe I have ever ran into a bash command in the past that didn't accept shorthand. IMO, it would be a good thing to support.

**Comment by johanburati** [comment]
I was running *azcopy* from a container I thought there was some permission issue with the storage, then I tried the some command directly on my desktop and got the same error.

It definitively should handle shorthand, don't know any other command that can't. 
And also it should also return a more descriptive error message, "runtime error: slice bounds out of range" is not great.

**Comment by zezha-msft** [comment]
Fixed in 10.1.0.
